ui/pages/institutional/manual-connect-custodian/manual-connect-custodian.tsx

```diff
@@ -0,0 +1,181 @@
+  handleConnectError: (e: Error) => void;
+  setAccounts: (accountsValue: object) => void;
+  removeConnectRequest: () => void;
+  connectRequest: object;
+};
+
+const ManualConnectCustodian: React.FC<ManualConnectCustodianProps> = ({
+  custodianImage,
+  custodianDisplayName,
+  jwtList = [],
+  token = '',
+  loading,
+  custodianName,
+  custodianType,
+  addNewTokenClicked,
+  connectRequest,
+  setCurrentJwt,
+  setConnectError,
+  handleConnectError,
+  setAccounts,
+  removeConnectRequest,
+  cancelConnectCustodianToken,
+}) => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const mmiActions = mmiActionsFactory();
+  const trackEvent = useContext(MetaMetricsContext);
+
+  return (
+    <>
+      <Box
+        padding={4}
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        className="page-container__content"
+        width={BlockSize.Full}
+      >
+        {window.innerWidth > 400 && (
+          <Box
+            display={Display.Flex}
+            alignItems={AlignItems.center}
+            marginBottom={4}
+            marginTop={4}
+          >
+            <ButtonIcon
+              data-testid="custody-back-button"
+              ariaLabel={t('back')}
+              iconName={IconName.ArrowLeft}
+              size={ButtonIconSize.Sm}
+              color={IconColor.iconDefault}
+              onClick={cancelConnectCustodianToken}
+              display={[Display.Flex]}
+            />
+            <Text>{t('back')}</Text>
+          </Box>
+        )}
+        {custodianImage && (
+          <Box display={Display.Flex} alignItems={AlignItems.center}>
+            <img
+              width={32}
+              height={32}
+              src={custodianImage}
+              alt={custodianDisplayName}
+            />
+            <Text as="h4" marginLeft={2}>
+              {custodianDisplayName}
+            </Text>
+          </Box>
+        )}
+        <Text marginTop={4}>
+          {t('enterCustodianToken', [custodianDisplayName])}
+        </Text>
+        <Box paddingBottom={7}>
+          <JwtUrlForm
+            jwtList={jwtList}
+            currentJwt={token}
+            onJwtChange={(jwt) => setCurrentJwt(jwt)}
+            jwtInputText={t('pasteJWTToken')}
+          />
+        </Box>
+      </Box>
+      <Box as="footer" className="page-container__footer" padding={4}>
+        {loading ? (
+          <PulseLoader />
+        ) : (
+          <Box display={Display.Flex} gap={4}>
+            <Button
+              data-testid="custody-cancel-button"
+              block
+              variant={ButtonVariant.Secondary}
+              size={ButtonSize.Lg}
+              onClick={cancelConnectCustodianToken}
+            >
+              {t('cancel')}
+            </Button>
+            <Button
+              block
+              data-testid="jwt-form-connect-button"
+              size={ButtonSize.Lg}
+              onClick={async () => {
```

Comment:
Nit: Can this onClick be moved out for cleaning code and easier reading purpose?